### PR TITLE
migration: create migration data resource

### PR DESCRIPTION
**Pull request by jma**
You can review the two commits together.

### CODE REVIEW

**Comment by PascalRepond** [comment]
## PO tests

- [x] REP: check conversion view: make two separate scrollers so that I can compare the data without having to scroll back and forth

File: rero_ils/modules/migrations/data/api.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+class IndexCfg:
+    """Migration Data Index configuration."""
+
+    settings = {"number_of_shards": 2, "number_of_replicas": 2}
+    aliases = {"migration-data": {}}
+
+
+class ConversionStatus(Enum):
+    """Class holding all available migration statuses."""
```
Comment (PascalRepond) [inline rero_ils/modules/migrations/data/api.py:34]: "data conversion status" and not migration status

File: tests/migrations/api/test_migrations_permissions.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            "search": False,
+            "read": False,
+            "create": False,
+            "update": False,
+            "delete": False,
+        },
+        migration,
+    )
+
+    # Librarian without allowed roles
```
Comment (PascalRepond) [inline tests/migrations/api/test_migrations_permissions.py:88]: with* 1 allowed role

File: tests/migrations/api/test_migrations_permissions.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+            "search": True,
+            "read": True,
+            "create": False,
+            "update": False,
+            "delete": False,
+        },
+        migration,
+    )
+
+    # Librarian without allowed roles
```
Comment (PascalRepond) [inline tests/migrations/api/test_migrations_permissions.py:106]: with* allowed roles

File: tests/migrations/api/test_migrations_permissions.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
```
Comment (PascalRepond) [inline tests/migrations/api/test_migrations_permissions.py:125]: with* 1 allowed role

**Review by PascalRepond — APPROVED** [review]

File: rero_ils/modules/migrations/cli.py [hunk 5]
```diff
@@ -48,6 +53,7 @@ def index():
 @with_appcontext
 def init():
     """Initialize the migration elasticsearch index."""
+    print(Migration._index.to_dict())
```
Comment (rerowep) [inline rero_ils/modules/migrations/cli.py:56]: Is this a left over print? If not use `click.echo`.

File: rero_ils/modules/migrations/data/cli.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    except NotFoundError:
+        click.secho(f"Migration with: {migration} does not exists", fg="yellow")
+        raise click.Abort()
+    DataMigration = migration.data_class
+
+    def display(data):
+        """Display the data for a given format."""
+        if format == "raw":
+            print(data.raw.decode())
+        if format == "json":
```
Comment (rerowep) [inline rero_ils/modules/migrations/data/cli.py:84]: Maybe use `elif`?

Comment (jma) [reply rero_ils/modules/migrations/data/cli.py:84]: I should not make any difference.

File: rero_ils/modules/imports/serializers/serializers.py [hunk 7]
```diff
@@ -148,31 +148,32 @@ def post_process(self, metadata):
 class ImportsMarcSearchSerializer(JSONSerializer):
     """Mixin serializing records as JSON."""
 
+    @classmethod
+    def sort_ordered_dict(cls, ordered_dict):
+        """."""
```
Comment (Garfield-fr) [inline rero_ils/modules/imports/serializers/serializers.py:153]: Missing docstring

File: rero_ils/modules/migrations/__init__.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2022 RERO
```
Comment (Garfield-fr) [inline rero_ils/modules/migrations/__init__.py:4]: 2024 ?

File: rero_ils/modules/migrations/api.py [hunk 9]
```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2022 RERO
```
Comment (Garfield-fr) [inline rero_ils/modules/migrations/api.py:4]: 2024 ?

File: rero_ils/modules/migrations/data/__init__.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2022 RERO
```
Comment (Garfield-fr) [inline rero_ils/modules/migrations/data/__init__.py:4]: 2024 ?

File: rero_ils/modules/migrations/data/api.py [hunk 11]
```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2022 RERO
```
Comment (Garfield-fr) [inline rero_ils/modules/migrations/data/api.py:4]: 2024 ?

File: rero_ils/modules/migrations/data/views.py [hunk 12]
```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2022 RERO
```
Comment (Garfield-fr) [inline rero_ils/modules/migrations/data/views.py:4]: 2024 ?

File: rero_ils/modules/migrations/permissions.py [hunk 13]
```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2022 RERO
```
Comment (Garfield-fr) [inline rero_ils/modules/migrations/permissions.py:4]: 2024 ?

File: rero_ils/modules/migrations/views.py [hunk 14]
```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2022 RERO
```
Comment (Garfield-fr) [inline rero_ils/modules/migrations/views.py:4]: 2024 ?

File: scripts/convert_marc21xml.py [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Mock Modules."""
```
Comment (Garfield-fr) [inline scripts/convert_marc21xml.py:18]: Is this comment correct?

File: tests/migrations/api/test_migration_data_rest.py [hunk 16]
```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2021-2023 RERO
```
Comment (Garfield-fr) [inline tests/migrations/api/test_migration_data_rest.py:4]: 2024

File: tests/migrations/api/test_migrations_permissions.py [hunk 17]
```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2022 RERO
```
Comment (Garfield-fr) [inline tests/migrations/api/test_migrations_permissions.py:4]: 2024 ?

File: tests/migrations/api/test_migrations_rest.py [hunk 18]
```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2021-2023 RERO
```
Comment (Garfield-fr) [inline tests/migrations/api/test_migrations_rest.py:4]: 2024 ?

File: tests/migrations/conftest.py [hunk 19]
```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2023 RERO
```
Comment (Garfield-fr) [inline tests/migrations/conftest.py:4]: 2024 ?

File: tests/migrations/ui/conftest.py [hunk 20]
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2023 RERO
```
Comment (Garfield-fr) [inline tests/migrations/ui/conftest.py:4]: 2024 ?

File: tests/migrations/ui/test_migration_data_api.py [hunk 21]
```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2021-2023 RERO
```
Comment (Garfield-fr) [inline tests/migrations/ui/test_migration_data_api.py:4]: 2024 ?

File: tests/migrations/ui/test_migrations_api.py [hunk 22]
```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2021-2023 RERO
```
Comment (Garfield-fr) [inline tests/migrations/ui/test_migrations_api.py:4]: 2024 ?

File: scripts/convert_marc21xml.py (hunk 15 above)
Comment (jma) [reply scripts/convert_marc21xml.py:18]: Yes.

**Pull request merged**